crates/protocol/src/info/variant.rs

```diff
@@ -373,6 +413,8 @@ mod test {
             blob_base_fee: 1,
             blob_base_fee_scalar: 810949,
             base_fee_scalar: 1368,
+            empty_scalars: false,
+            l1_fee_overhead: U256::ZERO,
```

Comment:
Can this be zero in the first ecotone block? If not, I don't see a need for field `empty_scalars`

Reply:
Also cc @clabby

Reply:
The first block of ecotone _always_ has empty scalars. This is because the `op-node` triggers the upgrade to the `L1Block` contract via a special deposit transaction, but when the L1 block info tx is sent for the first ecotone block, that upgrade has not happened yet (since this tx is always the first in the block.)

Reply:
It’s interesting then that revm decided to explicitly use this bool marker then. I’d prefer to leave the logic in revm as minimal diff as possible so this requires keeping this bool flag. If we remove it, we change the logic in revm which I’ve been working to avoid.

Reply:
defo agree, sorry I didn't pay more attention to which code is actually new